server/src/main/java/org/spine3/server/CommandService.java

```diff
@@ -0,0 +1,135 @@
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Sets;
+import io.grpc.stub.StreamObserver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.spine3.base.Command;
+import org.spine3.base.Response;
+import org.spine3.client.grpc.CommandServiceGrpc;
+import org.spine3.server.command.error.CommandException;
+import org.spine3.server.command.error.UnsupportedCommandException;
+import org.spine3.server.type.CommandClass;
+
+import java.util.Set;
+
+/**
+ * The {@code CommandService} allows client applications to post commands and
+ * receive updates from the application backend.
+ *
+ * @author Alexander Yevsyukov
+ */
+public class CommandService
+        extends CommandServiceGrpc.CommandServiceImplBase {
+
+    private final ImmutableMap<CommandClass, BoundedContext> boundedContextMap;
+
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    protected CommandService(Builder builder) {
+        this.boundedContextMap = builder.getBoundedContextMap();
+    }
+
+
+    @SuppressWarnings("RefusedBequest") // as we override default implementation with `unimplemented` status.
+    @Override
+    public void post(Command request, StreamObserver<Response> responseObserver) {
+        final CommandClass commandClass = CommandClass.of(request);
+        final BoundedContext boundedContext = boundedContextMap.get(commandClass);
+        if (boundedContext == null) {
+            handleUnsupported(request, responseObserver);
+        } else {
+            boundedContext.getCommandBus()
+                          .post(request, responseObserver);
+        }
+    }
+
+    private static void handleUnsupported(Command request, StreamObserver<Response> responseObserver) {
+        final CommandException unsupported = new UnsupportedCommandException(request);
+        log().error("Unsupported command posted to CommandService", unsupported);
+        responseObserver.onError(Statuses.invalidArgumentWithCause(unsupported));
+    }
+
+    public static class Builder {
+
+        private final Set<BoundedContext> boundedContexts = Sets.newHashSet();
+        private ImmutableMap<CommandClass, BoundedContext> boundedContextMap;
+
+
+        public Builder addBoundedContext(BoundedContext boundedContext) {
+            // Save it to a temporary set so that it is easy to remove it if needed.
+            boundedContexts.add(boundedContext);
+            return this;
+        }
+
+        public Builder removeBoundedContext(BoundedContext boundedContext) {
+            boundedContexts.remove(boundedContext);
```

Comment:
I think it is helpful to cover the removal for potential complex `CommandService.Builder` scenarios. 

Imagine you have a dynamic Java app, which Spine configuration is built on-the-fly depending on the system load. I.e. `I want the new node to serve BoundedContexts A and B. But not X and Y, since there are enough nodes covering those at the moment.`

In this case I imagine to have `addBoundedContext(...)` for A, B, X and Y in the base class like `AppBuilder` and `removeBoundedContext(...)` for X and Y  in descendants like `DistributedNodeBuilder extends AppBuilder`.